### DISCUSSION TITLE
Is it possible to get embeddings using this library?

### DISCUSSION THREAD

**Discussion by mloenow** [withcatai/node-llama-cpp#115] [Q&A] [answered]
Hi 👋

Awesome work!

Is it possible to get embeddings or is it not supported yet?

After digging around it looks like it's not the case 🤔

I basically want to be able to create a RAG supported solution, but can't get the embedding vectors for indexing without using llama.cpp directly or a different wrapper.

**Comment by giladgd** [comment]
@mloenow Getting embeddings is not supported at the moment.
Can you please give an example of how you may use it if it will be supported?

I aim to keep this library relatively high-level while still offering advanced capabilities, so understanding your use case can help me find the best way to add support for it, while maintaining a high-level approach.

**Comment by giladgd** [comment, marked as answer]
Update: I'll add support for this in one of the next beta versions

**Reply by mloenow** [reply]
Awesome!
